### ISSUE TITLE
USB Serial on XP Embedded

### ISSUE THREAD

**Issue by mskoczen** [chipKIT32/chipKIT32-MAX#466]
Between version 20130715 and 20140316 the USB Serial broke on XP Embedded Windows. I tested this with a Quick240, Fubarino SD, and another chipkit compatible board that all use USB Serial from the chip.

**Comment by EmbeddedMan** [comment]
I believe that commit 5c8e43a fixes this issue. Here is the summary of what was changed:

We now refrain from sending any USB data up to the PC until one or more of the following happens:
* The PC has sent some data to us or tries to read some data from us (i.e. an application has opened the part and sent some data out or is trying to read data in)
* The PC has activated a hardware handshaking line (i.e. an application has opened the port - most apps do this)
Once we are sending data to the PC, we will continue to do so until one of the following things happens:
* The PC has deactivated the hardware handshaking line (i.e. the app has closed the port)
* The PC has not asked for any data from us for 50ms (this indicates that there is no application consuming the data on the PC side, and the PC's buffer is full, and it doesn't want any more data)
I also bumped up the number of USB buffers to 16 (from 4), which seemed to help smooth things out somewhat

The test sketch that I posted above in this thread also passes with these changes applied.
